web-app/src/main/webapp/script.js

```diff
@@ -59,14 +62,28 @@ function showMessageOnInfoWindow(message, position, map, infoWindow) {
 window.onload = () => {
   // document.getElementById("report-form").style.display = "none";
   document.getElementById('report-button').addEventListener('click', showReportForm);
-  document.getElementById('submit-button').addEventListener('click', postUserReport);
+  document.getElementById('submit-button').addEventListener('click', geocodeAddress);
```

Comment:
nit: I think it is better to leave submit-button event listener function as postUserReport, and call geoCodeAddress in the postUserReport.